### PR TITLE
Feature dependencies upgrade

### PR DESCRIPTION
**Pull request by ClovisDj**
This `PR` upgrades the main dependencies of this package. Notably moving to `python==3.6.9`, `django==2.2.7` and `djangorestframework==3.10.3`.
These changes need to be merged and published before proceeding with the related `profiles` and `transmission` one.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -24,11 +24,11 @@ python = ">=3.6,<3.8"  # Compatible python versions must be declared here
...
 boto3 = "~1.9"
 cryptography = "~2.5"
-django = "^2.2.4"
+django = "~2.2.7"
 django-enumfields = "~0.10.0"
 django-influxdb-tagged-metrics = '==1.3.3'
-djangorestframework = "^3.9.1"
-djangorestframework-jsonapi = "~2"
+djangorestframework = "~3.10"
+djangorestframework-jsonapi = "~3.0.0"
```
Comment (ajhodges) [inline pyproject.toml]: Technically `~3` and `~3.0.0` are equivalent, but I think I'd prefer `~3` here

File: Dockerfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ## Base image with python and entrypoint scripts ##
 ## ============================================= ##
-FROM python:3.6.8-alpine3.9 AS base
+FROM python:3.6.9-alpine3.10 AS base
```
Comment (ajhodges) [inline Dockerfile]: Is the Alpine update necessary? I think Lucas and I previously decided that it was a good idea to stay ~1 minor version behind the bleeding edge Alpine version. I'd probably prefer to stay on 3.9 until 3.11 is released unless we have a good reason to make the switch.

File: src/shipchain_common/pagination.py [hunk 3]
```diff
@@ -14,10 +14,18 @@
...
 
-from rest_framework_json_api.pagination import PageNumberPagination
+from rest_framework_json_api.pagination import JsonApiPageNumberPagination
 
 
-class CustomResponsePagination(PageNumberPagination):
+class JsonResponsePagination(JsonApiPageNumberPagination):
+    """
+    Django Rest Framework JsonApi version 3.0.0 introduces JsonApiPageNumberPagination as base pagination class
+    with page query param: 'page[number]' that needs to be overridden here with proper value
```
Comment (ajhodges) [inline src/shipchain_common/pagination.py]: :+1: thanks for this documentation

File: src/shipchain_common/pagination.py [hunk 4]
```diff
@@ -14,10 +14,18 @@
 limitations under the License.
 """
 
-from rest_framework_json_api.pagination import PageNumberPagination
+from rest_framework_json_api.pagination import JsonApiPageNumberPagination
 
 
-class CustomResponsePagination(PageNumberPagination):
+class JsonResponsePagination(JsonApiPageNumberPagination):
```
Comment (ajhodges) [inline src/shipchain_common/pagination.py]: We lost some fidelity on what this class does with the rename to `JsonResponsePagination`. How about `JsonApiPagePagination`?

Comment (ClovisDj) [reply src/shipchain_common/pagination.py]: `JsonApiPagePagination` sound good to me :+1:

**Pull request merged**